sdk/nodejs/runtime/stack.ts

```diff
@@ -104,7 +104,17 @@ async function massage(prop: any, seenObjects: Set<any>): Promise<any> {
     }
 
     if (Output.isInstance(prop)) {
-        return await massage(await prop.promise(), seenObjects);
+        // If the output itself is a secret, we don't want to lose the secretness by returning the underlying
+        // value. So instead, we massage the underlying value and then wrap it back up in an Output which is
+        // marked as secret.
+        const isSecret = await (prop.isSecret || Promise.resolve(false));
```

Comment:
when would isSecret be not there?

Reply:
If the `Output` instance that is getting exported was from an older SDK which did not contain the `isSecret` property, which could happen if we have multiple `@pulumi/pulumi`'s running around.